Multisig. Can't set call data for xcm operation

## Description

User can't set call data to XCM operation, error:

```
Start parsing call data for section polkadotXcm and method limitedTeleportAssets
multisigTxService.ts:224 Error during update callData:  TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at Object.parseXcmPalletExtrinsic (xcmService.ts:245:34)
    at polkadotxcm_limited_teleport_assets (callDataDecoder.ts:228:37)
    at getDecodedTransaction (callDataDecoder.ts:170:12)
    at parseSingle (callDataDecoder.ts:127:12)
    at decodeCallData (callDataDecoder.ts:66:12)
    at updateCallData (multisigTxService.ts:220:27)
    at setupCallData (OperationFullInfo.tsx:66:5)
    at submitHandler (CallDataModal.tsx:46:5)
    at react-hook-form.js?v=f13ddb07:1671:15
```

## Test data

[spektr-database.json](https://github.com/user-attachments/files/18621087/spektr-database.json)

## Steps to Reproduce

1. Import test data
2. Go to Operations
3. Try to set call data:
0x1f0904010100a10f04000101007a28037947ecebe0dd86dc0e910911cb33185fd0714b37b75943f67dcf9b6e7c04040000001300307ebabe0472020000000000

## Expected Behavior
Success parsed

## Actual Behavior
Failed

## Environment
https://github.com/novasamatech/nova-spektr/commit/1d871750621c741cf1e0a43f2889c37d1b2c83b8

--- a/src/renderer/features/operations/OperationsValidation/lib/transfer-rules.ts
+++ b/src/renderer/features/operations/OperationsValidation/lib/transfer-rules.ts
@@ -64,7 +64,7 @@ export const TransferRules = {
       validator: (destination: string, _: any, chain: Chain) => {
         if (!chain) return false;
 
-        return chain && validateAddress(destination, chain);
+        return validateAddress(destination, chain);
       },
     }),
   },

--- a/src/renderer/features/operations/OperationsValidation/lib/transfer-rules.ts
+++ b/src/renderer/features/operations/OperationsValidation/lib/transfer-rules.ts
@@ -62,8 +62,6 @@ export const TransferRules = {
       source,
       // Second argument for validator is form data, but we need chain
       validator: (destination: string, _: any, chain: Chain) => {
-        console.log(chain, destination);
-
         if (!chain) return false;
 
         return chain && validateAddress(destination, chain);

--- a/src/renderer/features/operations/OperationsValidation/lib/transfer-rules.ts
+++ b/src/renderer/features/operations/OperationsValidation/lib/transfer-rules.ts
@@ -1,7 +1,7 @@
 import { BN } from '@polkadot/util';
 import { type Store } from 'effector';
 
-import { type Account } from '@/shared/core';
+import { type Account, type Chain } from '@/shared/core';
 import { formatAmount, validateAddress } from '@/shared/lib/utils';
 import {
   type BalanceMap,
@@ -56,12 +56,19 @@ export const TransferRules = {
       errorText: 'transfer.requiredRecipientError',
       validator: Boolean,
     },
-    incorrectRecipient: {
+    incorrectRecipient: (source: Store<Chain | null>) => ({
       name: 'incorrectRecipient',
       errorText: 'transfer.incorrectRecipientError',
+      source,
       // Second argument for validator is form data, but we need chain
-      validator: (destination: string) => validateAddress(destination),
-    },
+      validator: (destination: string, _: any, chain: Chain) => {
+        console.log(chain, destination);
+
+        if (!chain) return false;
+
+        return chain && validateAddress(destination, chain);
+      },
+    }),
   },
   amount: {
     required: {

--- a/src/renderer/shared/api/xcm/service/xcmService.ts
+++ b/src/renderer/shared/api/xcm/service/xcmService.ts
@@ -241,7 +241,7 @@ function parseXcmPalletExtrinsic(args: Omit<XcmPalletTransferArgs, 'feeAssetItem
   const destJunction = Object.keys(destInterior)[0];
   parsedPayload.destParachain = Number(xcmUtils.toRawString(get(destInterior, `${destJunction}.Parachain`)));
 
-  if (!parsedPayload.isRelayToken) {
+  if (!parsedPayload.isRelayToken && assetInterior) {
     const assetJunction = Object.keys(assetInterior)[0];
     const cols = xcmUtils.getJunctionCols<{ GeneralIndex: string; Parachain: number }>(assetInterior, assetJunction);
 

--- a/src/renderer/widgets/Transfer/model/form-model.ts
+++ b/src/renderer/widgets/Transfer/model/form-model.ts
@@ -105,6 +105,18 @@ const $totalFee = combine(
   ({ fee, deliveryFee }) => new BN(fee).add(deliveryFee).toString(),
 );
 
+const $xcmChain = combine(
+  {
+    chains: networkModel.$chains,
+    xcmChainId: xcmTransferModel.$xcmChainId,
+  },
+  ({ chains, xcmChainId }) => {
+    if (!xcmChainId) return null;
+
+    return chains[xcmChainId] ?? null;
+  },
+);
+
 const $transferForm = createForm<FormParams>({
   fields: {
     account: {
@@ -138,7 +150,7 @@ const $transferForm = createForm<FormParams>({
     },
     destination: {
       init: '',
-      rules: [TransferRules.destination.required, TransferRules.destination.incorrectRecipient],
+      rules: [TransferRules.destination.required, TransferRules.destination.incorrectRecipient($xcmChain)],
     },
     amount: {
       init: '',
@@ -430,18 +442,6 @@ const $extrinsic = combine(
     if (!api || !coreTx) return null;
 
     return getExtrinsic[coreTx.type](coreTx.args, api);
-  },
-);
-
-const $xcmChain = combine(
-  {
-    chains: networkModel.$chains,
-    xcmChainId: xcmTransferModel.$xcmChainId,
-  },
-  ({ chains, xcmChainId }) => {
-    if (!xcmChainId) return null;
-
-    return chains[xcmChainId] ?? null;
   },
 );
 

--- a/src/renderer/widgets/Transfer/ui/TransferForm.tsx
+++ b/src/renderer/widgets/Transfer/ui/TransferForm.tsx
@@ -199,14 +199,14 @@ const Destination = () => {
   const { t } = useI18n();
 
   const {
-    fields: { destination },
+    fields: { destination, xcmChain },
   } = useForm(formModel.$transferForm);
 
   const isMyselfXcmEnabled = useUnit(formModel.$isMyselfXcmEnabled);
 
   const prefixElement = (
     <div className="flex h-auto items-center">
-      {validateAddress(destination.value) ? (
+      {validateAddress(destination.value, xcmChain.value) ? (
         <Identicon size={20} address={destination.value} background={false} />
       ) : (
         <Icon size={20} name="emptyIdenticon" />

